File: 3487-maximum-unique-subarray-sum-after-deletion/3487-maximum-unique-subarray-sum-after-deletion.py
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        N = len(nums)
        hashSet = set()
        total = 0 

        for i in range(N):
            if nums[i] in hashSet or nums[i] <= 0:
                continue
            total += nums[i]
            hashSet.add(nums[i])
        
        if total == 0 and N > 0:
            total = max(nums)
        
        return total
        

File: 3487-maximum-unique-subarray-sum-after-deletion/3487-maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> hashSet;
        int total = 0;

        for (int i = 0; i < n; ++i){
            if (hashSet.count(nums[i]) || nums[i] <= 0){
                continue;
            }
            total += nums[i];
            hashSet.insert(nums[i]);
        }
        
        if (total == 0 && n > 0){
            total = *std::max_element(nums.begin(), nums.end());
        }
        return total;
    }
};